micrometer-observation/src/main/java/io/micronaut/micrometer/observation/DefaultObservedFactory.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.micrometer.observation;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.observation.DefaultMeterObservationHandler;
+import io.micrometer.observation.GlobalObservationConvention;
+import io.micrometer.observation.ObservationFilter;
+import io.micrometer.observation.ObservationHandler;
+import io.micrometer.observation.ObservationPredicate;
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.tracing.Tracer;
+import io.micrometer.tracing.handler.TracingAwareMeterObservationHandler;
+import io.micrometer.tracing.propagation.Propagator;
+import io.micrometer.tracing.handler.DefaultTracingObservationHandler;
+import io.micrometer.tracing.handler.PropagatingReceiverTracingObservationHandler;
+import io.micrometer.tracing.handler.PropagatingSenderTracingObservationHandler;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Factory for Micrometer Observation integration. Creates the following beans:
+ *
+ * <ul>
+ *     <li>{@link ObservationRegistry} - Instrumented ObservationRegistry</li>
+ *     <li>{@link ObservationHandler} - Observation Handlers for exporting metrics, traces</li>
+ *     <li>{@link ObservationFilter} - ObservationFilter that adds common low cardinality key values to all observations</li>
+ * </ul>
+ */
+@Factory
+@Internal
+public final class DefaultObservedFactory {
+    @Singleton
+    ObservationRegistry observationRegistry(
+        List<ObservationPredicate> observationPredicates,
+        List<GlobalObservationConvention<?>> observationConventions,
+        List<ObservationHandler<?>> observationHandlers,
+        List<ObservationFilter> observationFilters
+    ) {
+        ObservationRegistry observationRegistry = ObservationRegistry.create();
+        observationHandlers.forEach(observationRegistry.observationConfig()::observationHandler);
+        observationPredicates.forEach(observationRegistry.observationConfig()::observationPredicate);
+        observationFilters.forEach(observationRegistry.observationConfig()::observationFilter);
+        observationConventions.forEach(observationRegistry.observationConfig()::observationConvention);
+        return observationRegistry;
+    }
+
+    @Singleton
+    @Requires(classes = Tracer.class)
+    @Requires(bean = Tracer.class)
+    public ObservationHandler<?> defaultTracingObservationHandler(Tracer tracer) {
```

Comment:
This has to be last in the `TracingObservationHandler` list AFTER `sender` and `receiver`